ngraph/core/src/op/interpolate.cpp

```diff
@@ -111,143 +114,536 @@ namespace ngraph
         visitor.on_attribute("pads_end", m_ref.pads_end);
         return true;
     }
-}
+} // namespace ngraph
 
-// Interpolate v3
+// Interpolate v4
 
-constexpr NodeTypeInfo op::v3::Interpolate::type_info;
+NGRAPH_RTTI_DEFINITION(op::v4::Interpolate, "Interpolate", 4);
 
-op::v3::Interpolate::Interpolate(const Output<Node>& image,
+op::v4::Interpolate::Interpolate(const Output<Node>& image,
                                  const Output<Node>& output_shape,
-                                 const op::v3::Interpolate::InterpolateAttrs& attrs)
-    : Op({image, output_shape})
+                                 const Output<Node>& scales,
+                                 const Output<Node>& axes,
+                                 const op::v4::Interpolate::InterpolateAttrs& attrs)
+    : Op({image, output_shape, scales, axes})
+    , m_attrs(attrs)
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v4::Interpolate::Interpolate(const Output<Node>& image,
+                                 const Output<Node>& output_shape,
+                                 const Output<Node>& scales,
+                                 const op::v4::Interpolate::InterpolateAttrs& attrs)
+    : Op({image, output_shape, scales})
     , m_attrs(attrs)
 {
     constructor_validate_and_infer_types();
 }
 
-bool op::v3::Interpolate::visit_attributes(AttributeVisitor& visitor)
+bool op::v4::Interpolate::visit_attributes(AttributeVisitor& visitor)
 {
     visitor.on_attribute("attrs", m_attrs);
     return true;
 }
 
-void op::v3::Interpolate::validate_and_infer_types()
+static const char* cannot_define_axes =
+    "Could not define axes of interpolation because there"
+    "are only three inputs and input data has a dynamic rank.";
+
+std::vector<int64_t> op::v4::Interpolate::get_axes() const
 {
-    NODE_VALIDATION_CHECK(this,
-                          get_input_element_type(1).is_integral_number(),
-                          "output shape must be an integral number.");
-    set_input_is_relevant_to_shape(1);
+    auto inputs = input_values();
+    if (inputs.size() <= 3)
+    {
+        PartialShape input_shape = PartialShape(get_input_partial_shape(0));
+        NODE_VALIDATION_CHECK(this, !input_shape.rank().is_dynamic(), cannot_define_axes);
```

Comment:
We usually check the "positive case" when validating a node. Can you switch from `!input_shape.rank().is_dynamic()` to `input_shape.rank().is_static()`?

Reply:
Fixed.